lib/ctblsolv.gi

```diff
@@ -11,6 +11,65 @@
 ##  This file contains character table methods for solvable groups.
 ##
 
+##  We take advantage of knowing conjugacy class representatives
+##  as words in generators.
+InstallMethod(LinearCharacters, ["CanEasilyComputePcgs"], function(G)
+  local pcgs, D, hom, Gab, abinv, exp, e, Ee, genexp, 
+        clexps, tab, irgens, a, lin, c, res, j, i, sz;
+  if Size(G) = 1 then
+    return [TrivialCharacter(G)];
+  fi;
+  pcgs := Pcgs(G);
+  D := DerivedSubgroup(G);
+  hom := NaturalHomomorphismByNormalSubgroupNC(G, D);
```

Comment:
I was not aware of `MaximalAbelianQuotient`, I would never have guessed that this operation would return a homomphism (but its image). Since the method for `MaximalAbelianQuotient` just returns `NaturalHomomorphismByNormalSubgroupNC(G, DerivedGroup(G))` I prefer to omit this wrapper.

Reply:
Concerning the name `MaximalAbelianQuotient`, I have opened the issue #4913.
Concerning the decision not to call `MaximalAbelianQuotient`, I do not catch the point. The code needs the epimorphism but not the derived subgroup, thus `MaximalAbelianQuotient( G )` is exactly what is needed.

Reply:
Good idea, to open a issue about these confusing names, I had not noticed before that this is more general problem.
The argument that `MaximalAbelianQuotient` is an attribute convinced me to change the code.
I had assigned `D` because I used it in several comments - now using `G'` in the comments.